tests/instruments/hp/test_hp8753e_with_device.py

```diff
@@ -0,0 +1,384 @@
+#
+# This file is part of the PyMeasure package.
+#
+# Copyright (c) 2013-2023 PyMeasure Developers
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+import logging
+from contextlib import contextmanager
+from time import sleep
+
+import numpy as np
+import pytest
+from pyvisa.errors import VisaIOError
+
+from pymeasure.adapters import PrologixAdapter
+from pymeasure.instruments.hp import HP8753E
+
+# pytest.skip("Only works with connected hardware", allow_module_level=True)
+
+
+@contextmanager
+def init_prologix_adapter():
+    try:
+        prologix = PrologixAdapter(
+            resource_name="ASRL4::INSTR", address=16, visa_library="@py", auto=1
```

Comment:
Instead of hardcoding the resource_name, we have a fixture ready, see the [device-tests documentation](https://pymeasure.readthedocs.io/en/latest/dev/adding_instruments/tests.html#device-tests).

Reply:
I'm not sure how to use that to pass both the PrologixAdapter address as well as the GPIB address.

Working on creating unit tests for my Agilent E4418B I kludged this workaround:

    # use this command to run the test without editing it
    # for prologix adapters
    # pytest '.\tests\instruments\agilent\test_agilentE4418B_with_device.py' --device-address="PrologixAdapter,ASRL4::INSTR,13"
    # everyone else, the `--device-address` flag works normally
    # pytest '.\tests\instruments\agilent\test_agilentE4418B_with_device.py' --device-address="GPIB::whatever::your::visa::says"
    
    
    @pytest.fixture(scope="module")
    def power_meter(connected_device_address):
        try:
            if "PrologixAdapter" not in connected_device_address:
                power_meter = AgilentE4418B(connected_device_address)
            else:
                _, prologix_address, gpib_address, *other_address_info = connected_device_address.split(",")
                prologix = PrologixAdapter(resource_name=prologix_address, visa_library="@py", auto=1)
                # need to ensure `eot_enable` is set to zero otherwise you will have to read twice to
                # get rid of the extra new line character
                prologix.write("++eot_enable 0")
                power_meter = AgilentE4418B(adapter=prologix.gpib(int(gpib_address)))
        except IOError:
            print("Not able to connect to power meter")
            assert False
    
        yield power_meter

Reply:
I'll wait for feedback on how to read the value from `pytest --device-address` and see if there is an appropriate way to pass all of the necessary information to use the `PrologixAdapter` in the fixture.

Reply:
There is no support (yet) to use a prologix adapter in these tests.

We could modify the fixture to generate a prologix adapter if needed.

Reply:
Let me know if this can stay as is or what to change it to. Since I have the Prologix adapter, I can't use the test fixture to code it.

Reply:
Cool idea to use the Prologix in the resource-name!